src/shims/fs.rs

```diff
@@ -1,16 +1,21 @@
 use std::collections::HashMap;
-use std::convert::TryFrom;
+use std::convert::{TryInto, TryFrom};
 use std::fs::{remove_file, File, OpenOptions};
 use std::io::{Read, Write};
+use std::path::PathBuf;
+use std::time::SystemTime;
 
-use rustc::ty::layout::{Size, Align};
+use rustc::ty::layout::{Size, Align, LayoutOf};
 
 use crate::stacked_borrows::Tag;
 use crate::*;
+use helpers::immty_from_uint_checked;
+use shims::time::system_time_to_duration;
 
 #[derive(Debug)]
 pub struct FileHandle {
     file: File,
+    path: PathBuf,
```

Comment:
Is this new field even still used now?

Reply:
I don't think so. We removed the untestable cases of `statx` with the weird combinations of file descriptor and path.